### PR TITLE
refactor: Clean state when fails enabling/connecting wallet without dispatching disconnectWallet

### PR DESCRIPTION
**Pull request by kevinszuchet**
Related to https://github.com/decentraland/marketplace/issues/1619

### CODE REVIEW

File: src/modules/wallet/sagas.spec.ts [hunk 1]
```diff
@@ -1,25 +1,113 @@
...
+        return expectSaga(walletSaga)
+          .provide([
+            [
+              race({
+                success: take(FETCH_WALLET_SUCCESS),
+                failure: take(FETCH_WALLET_FAILURE)
+              }),
+              { success: null, failure: { payload: { error } } }
+            ]
+          ])
```
Comment (LautaroPetaccio) [inline src/modules/wallet/sagas.spec.ts:47]: What if we mock, in the provide section, the `put(fetchWalletRequest())` so the action doesn't get handled?
The same could be applied to the other tests of the saga that put the same action.

**Review by LautaroPetaccio — COMMENTED** [review]
Looks great ✅ I've just left a small comment on a test.

**Review by LautaroPetaccio — APPROVED** [review]
Send it 🚀

**Review by cazala — APPROVED** [review]
LGTM ✅

**Pull request merged**